app/services/usda_service.rb

```diff
@@ -0,0 +1,19 @@
+class UsdaService
+
+   def self.get_url
+      Faraday.new("https://api.nal.usda.gov") do |faraday|
+        faraday.params["api_key"] = ENV['usda_key']
+      end
+    end
+
+    def self.name_search(name)
+      @_name_search = JSON.parse(get_url.get("/fdc/v1/foods/search?query=#{name}").body, symbolize_names: true)
```

Comment:
I think you could make a single call here and just add params for pageSize = 10. This would give you all the info you need in a single method.